### PR TITLE
chore: reviewdog pmd

### PR DESCRIPTION
**Pull request by Meemaw**

### CODE REVIEW

File: .github/workflows/backend:check.yml [hunk 1]
```diff
@@ -24,11 +24,24 @@ jobs:
           arguments: pmdMain pmdTest --parallel --continue
           build-root-directory: backend
           wrapper-directory: backend
-      - uses: actions/upload-artifact@v2
-        if: failure()
+      - name: Merge reports
+        run: |
+          sudo apt-get install zsh
+          head -n 5 backend/auth/auth-api/build/reports/pmd/main.xml > report.xml
+          zsh -c 'for file in backend/**/build/reports/pmd/main.xml;do;cat "$file" | tail -n +6 | sed "$ d" >> report.xml;done'
```
Comment (snuderl) [inline .github/workflows/backend:check.yml]: za to se res rabi `zsh`?

**Review by snuderl — DISMISSED** [review]
looks cool

File: .github/workflows/backend:check.yml (hunk 1 above)
Comment (Meemaw) [reply .github/workflows/backend:check.yml]: I couldn't get globstar (**) to work in bash. I would prefer if we could do that without `zsh` as installation takes ~30s.

Comment (snuderl) [reply .github/workflows/backend:check.yml]: try bash -c `shopt -s globstar && ...`

Comment (Meemaw) [reply .github/workflows/backend:check.yml]: Will merge this as this PR makes it work and will try with bash in a follow up PR.

**Pull request merged**